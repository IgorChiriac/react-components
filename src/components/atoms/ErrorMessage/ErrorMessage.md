Error message for input Components. This component is meant to be used with an input.

For a more complex component check `TextField`.

    <ErrorMessage>Oops ! There has been an error !</ErrorMessage>
