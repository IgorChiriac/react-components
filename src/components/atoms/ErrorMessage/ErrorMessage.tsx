import React, { FC } from 'react';
import styled from 'styled-components';
import { colors } from '../../../constants/colors';
import warningIcon from '../../../content/images/red-warning.svg';
import Text from '../Text/Text';

const ErrorMessage = styled(Text).attrs({
  role: 'alert',
  weight: 'semibold',
  forwardedAs: 'p',
  size: 'small',
})`
  width: 100%;
  color: ${colors.semantic.error};
  display: block;

  &:before {
    content: '';
    background: url(${warningIcon}) no-repeat;
    width: 28px;
    height: 16px;
    display: inline-block;
    vertical-align: middle;
  }
`;

// TODO: Styleguidist to be able to locate styled components. See #147.
export const StyleguidistErrorMessage: FC<{}> = props => <ErrorMessage {...props} />;

export default ErrorMessage;
