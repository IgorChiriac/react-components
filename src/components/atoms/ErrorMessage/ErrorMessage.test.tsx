import { axe } from 'jest-axe';
import React from 'react';
import { render } from 'react-testing-library';
import ErrorMessage from './ErrorMessage';

describe('<ErrorMessage />', () => {
  it('renders the component with props with no a11y violations', async () => {
    const { container } = render(<ErrorMessage>Ops error !</ErrorMessage>);
    expect(container.firstChild).toMatchSnapshot();
    const results = await axe(container.innerHTML);
    expect(results).toHaveNoViolations();
  });
});
