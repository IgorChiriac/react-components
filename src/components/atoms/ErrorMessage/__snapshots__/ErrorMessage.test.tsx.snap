// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ErrorMessage /> renders the component with props with no a11y violations 1`] = `
.c1 {
  margin: 0;
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  color: #0D0A38;
  line-height: 1.6;
  font-family: "Open Sans",-apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-weight: 600;
  font-size: 16px;
}

.c0 {
  width: 100%;
  color: #EE0505;
  display: block;
}

.c0:before {
  content: '';
  background: url(red-warning.svg) no-repeat;
  width: 28px;
  height: 16px;
  display: inline-block;
  vertical-align: middle;
}

<span
  class="c0 c1"
  role="alert"
>
  Ops error !
</span>
`;
